dotcom-rendering/src/web/lib/addFontsToIframe.ts

```diff
@@ -0,0 +1,71 @@
+// The ts.dom interface for FontFaceSet does not contain the .add method
+type FontFaceSetWithAdd = FontFaceSet & {
+	add?: { (font: FontFace): void };
+};
+
+/**
+ * clone the whole style element containing the `font-face` declarations
+ * from the parent and append it to the iframe's document head.
+ */
+const cloneFontFaceElement = (
+	parentDocument: Document,
+	iframe: HTMLIFrameElement,
+) => {
+	const webfontStyleElementOnParent =
+		parentDocument.querySelector('style.webfont');
+	if (!webfontStyleElementOnParent) {
+		return;
+	}
+	const clonedStyleElement = webfontStyleElementOnParent.cloneNode(true);
+	iframe.contentWindow?.document.head.appendChild(clonedStyleElement);
+};
+
+/**
+ * Copy a subset of fonts from the parent document to an iframe.
+ * If the browser does not support the FontFace.add method,
+ * clones the style element with the "webfont" class and appends
+ * it to the iframe's document head.
+ */
+export const addFontsToIframe = (
+	parentDocument: Document,
+	iframe: HTMLIFrameElement,
+	requiredFontNames: string[],
+): void => {
+	// FontFace.add is not supported (IE), clone the element instead
+	const iframeFontFaceSet = iframe.contentDocument?.fonts as
+		| undefined
+		| FontFaceSetWithAdd;
+	if (!iframeFontFaceSet?.add) {
+		cloneFontFaceElement(parentDocument, iframe);
+		return;
+	}
+
+	// get all the fontFaces on the parent matching the list of font names
+	const requiredFonts: FontFace[] = [];
+	parentDocument.fonts.forEach((fontFace) => {
+		if (requiredFontNames.includes(fontFace.family)) {
+			requiredFonts.push(fontFace);
+		}
+	});
```

Comment:
```suggestion
	const requiredFonts: FontFace[] = parentDocument.fonts.filter((fontFace) => requiredFontNames.includes(fontFace.family)));
```

Nice spot for a filter maybe?